pkg/scaling/scale_handler.go

```diff
@@ -312,7 +315,7 @@ func (h *scaleHandler) buildScalers(ctx context.Context, withTriggers *kedav1alp
 			if scaler != nil {
 				scaler.Close(ctx)
 			}
-			continue
+			return result, err
```

Comment:
I am not sure whether we shouldn't handle closing of all other scalers in here. @ahmelsayed WDYT?

Reply:
maybe something like this:
```golang
	for _, builder := range result {
		builder.Scaler.Close(ctx)
	}
	return nil, err
```
I think that we could close others because the cache will be refreshed in the worst case.
WDYT?